sunscreen_runtime/src/debugger/server.rs

```diff
@@ -0,0 +1,297 @@
+use actix_web::{get, http::header, web, App, HttpResponse, HttpServer, Responder};
+
+use semver::Version;
+
+use crate::{
+    debugger::sessions::Session,
+    debugger::{decrypt_seal, get_mult_depth, overflow_occurred},
+    debugger::{get_sessions, BfvNodeType, DebugNodeType, ZkpNodeType},
+    Ciphertext, InnerCiphertext, InnerPlaintext, Plaintext, Runtime, SealData, Type, WithContext,
+};
+use petgraph::stable_graph::NodeIndex;
+use std::sync::OnceLock;
+use std::thread;
+
+use tokio::runtime::Builder;
+
+static SERVER: OnceLock<()> = OnceLock::new();
+
+#[cfg(feature = "debugger")]
+/**
+ * Lazily starts a webserver at `127.0.0.1:8080/`.
+ */
+pub fn start_web_server() {
+    SERVER.get_or_init(|| {
+        thread::Builder::new()
+            .name("debugger".to_owned())
+            .spawn(|| {
+                let rt = Builder::new_current_thread().enable_all().build().unwrap();
+
+                rt.block_on(async {
+                    HttpServer::new(move || {
+                        App::new()
+                            .service(get_session_data)
+                            .service(get_all_sessions)
+                            .service(get_code)
+                            .service(get_node_data)
+                            .service(get_stack_trace)
+                            .service(index)
+                            .service(app_css)
+                            .service(main_js)
+                    })
+                    .disable_signals()
+                    .bind(("127.0.0.1", 8080))
+                    .unwrap()
+                    .run()
+                    .await
+                    .unwrap()
+                });
+            })
+            .unwrap();
+    });
+}
+
+#[cfg(feature = "debugger")]
```

Comment:
You don't need to gate these, as the entire module it lives in is gates (I think)